### PR TITLE
feat: Add fetching of nft output and data on nft details page by nftId

### PR DESCRIPTION
**Pull request by abbasabid**
# Description of change

Added fetching of NFT data on NFT details page.
Aims to complete https://github.com/iotaledger/explorer/issues/267

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

It should probably work but don't know how to test it. Currently iota.js is not able to create a transaction for nft and there are not valid nftId's on the network.

### CODE REVIEW

File: client/src/app/App.tsx [hunk 1]
```diff
@@ -286,7 +286,7 @@ class App extends Component<RouteComponentProps<AppRouteProps> & { config: IConf
                                                 )}
                                             />
                                             <Route
-                                                path="/:network/nft/:tokenId"
+                                                path="/:network/nft/:nftId"
                                                 component={(props: RouteComponentProps<NFTDetailsRouteProps>) =>
                                                 (
                                                     <NFTDetails
```
Comment (msarcev) [inline client/src/app/App.tsx:292]: Please add a check here to this works only if network `isStardust` (look above). If not dont render anything. Actually add it around the route so the route will exist only if currently on stardust network.

File: client/src/app/routes/stardust/NFTDetails.tsx [hunk 2]
```diff
@@ -293,6 +307,27 @@ class NFTDetails extends AsyncComponent<RouteComponentProps<NFTDetailsRouteProps
...
+            nftId
+        });
+
+        if (nftOutputs?.outputs && nftOutputs?.outputs?.items.length > 0) {
+            for (const outputId of nftOutputs.outputs.items) {
+                const output = await this._tangleCacheService.outputDetails(networkId, outputId);
+
+                if (output && !output.isSpent && output.output.type === NFT_OUTPUT_TYPE) {
+                    const nftDetails = output.output;
+                    // TODO: Code here to map the data on the page
```
Comment (msarcev) [inline client/src/app/routes/stardust/NFTDetails.tsx:325]: Missing something here ?

**Review by msarcev — CHANGES_REQUESTED** [review]

File: client/src/app/App.tsx (hunk 1 above)
Comment (abbasabid) [reply client/src/app/App.tsx]: done

Comment (abbasabid) [reply client/src/app/App.tsx]: done

File: client/src/app/routes/stardust/NFTDetails.tsx (hunk 2 above)
Comment (abbasabid) [reply client/src/app/routes/stardust/NFTDetails.tsx]: Map and displayed amount and nftId which we get in nftDetails object.

Comment (abbasabid) [reply client/src/app/routes/stardust/NFTDetails.tsx]: Map and displayed amount and nftId which we get in nftDetails object.

**Review by msarcev — APPROVED** [review]

**Pull request merged**